docs/geps/gep-01.rst

```diff
@@ -0,0 +1,148 @@
+===========================
+GEP 1 - Variable Names
+===========================
+
+:Author: `Eric Sommer <https://github.com/Eric-Sommer>`
+:Status: Provisional
+:Type: Standards
+:Created: 2019-11-04
+
+
+This GEP specifies a framework on how to name variables. This affects
+1. input variables: what the user feeds into the model
+2. intermediate variables: variables needed for the tax-benefit model, but which are not returned by default.
+3. output variables
+
+A convention on variable names should make the code easier to read and to maintain. In addition, it should lower the requirement to read the documentation (once there is one).
+
+General Guidelines
+-----------------
+- in principal, use lower case letters only. Exceptions are specified below.
+- longer names might be interrupted with underscores, e.g. `alg2_anzurechnendes_ek`
+- use of booelan type for dummy variables rather than integer. Boolean variables can be negated easily (mind the difference between `~` and `not`) and make the code easier to read.
+
+Input Variables
+-----------------
+
+First letter specifies type of variable
+----------------------------------------
+In the spirit of the Euromod_ framework, I suggest that the type of a variable is indicated by its first letter
```

Comment:
Thanks for all your work on this. 

Personally, I am not a big fan of the EUROMOD approach, which I would summarise as "Make the variable names incomprehensibly short, but easy to look up." I can see how this will grow on you if you work with it for sufficiently long time, but our goal is precisely to draw in people who do not.

My suggestion would be closer to what @janosg describes [here](https://estimagic.readthedocs.io/en/latest/contributing/styleguide.html). We are a Python package, after all and should try to be pythonic. Though we have [this](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds), of course ...

Reply:
Very valid points. The reason I started with (not copied) the EUROMOD approach is that, while having worked with a few survey datasets, the Euromod approach is the one where, at one point, you stop consulting the codebook every time. This never happened with SOEP to me. I'm not at all against longer, self-speaking names, while I see a lot of use in a categorical first letter and the last letter indicating the time reference.